mkt/webapps/migrations/0013_disable_unsupported_apps.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations
+from mkt.constants.applications import DEVICE_GAIA
+from mkt.constants.base import ADDON_FREE, STATUS_NULL
+
+
+def disable_unsupported_apps(apps, schema_editor):
+    Webapp = apps.get_model("webapps", "Webapp")
+    Webapp.objects.exclude(addondevicetype__device_type=DEVICE_GAIA.id,
+                           premium_type=ADDON_FREE).update(status=STATUS_NULL)
```

Comment:
I think something like this would work ? `Webapp.objects.only('id').exclude(Q(addondevicetype__device_type=DEVICE_GAIA.id) | Q(premium_type=ADDON_FREE))`

Or just make 2 queries maybe, so it's easier to read ?